recipes/libvdwxc/meta.yaml

```
@@ -0,0 +1,62 @@
+{% set name = "libvdwxc" %}
+{% set version = '0.4.0' %}
+{% set build_number = 0 %}
+
+{% if mpi != 'nompi' %}
+  {% set mpi_prefix = "mpi_" + mpi %}
+{% else %}
+  {% set mpi_prefix = "nompi" %}
+{% endif %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://launchpad.net/{{ name }}/stable/{{ version }}/+download/{{ name }}-{{ version }}.tar.gz
+  sha256: 3524feb5bb2be86b4688f71653502146b181e66f3f75b8bdaf23dd1ae4a56b33
+
+build:
+  skip: True  # [not unix]
+  number: {{ build_number }}
+  string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build_number }}"
+  run_exports:
+    {% if mpi != 'nompi' %}
+    - {{ name }} * {{ mpi_prefix }}_*
+    {% endif %}
```

Are run exports not needed when using the `nompi` variant?
```suggestion
  run_exports:
    {% if mpi != 'nompi' %}
    - {{ name }} * {{ mpi_prefix }}_*
    {% endif %}
    - {{ pin_subpackage('libvdwxc', max_pin='x.x') }}
```

---

From my (still) limited understanding of run_exports, it looks like the condition is not necessary indeed. I copied what had been done in other similar packages. I suppose the main concern of those other packages was not mixing mpi implementations and if the 'nompi' version is the default anyway, all should work. **But**, if removing the condition just make things more reliable, I don't see why not removing it.

---

Anyway, I have removed it.

---

> From my (still) limited understanding of run_exports

run_exports tells conda that "If libvdwxc is in the build/host section, then you need to put libvdwxc in your run requirements too" because any package that links against libvdwc needs to have it at runtime. (As opposed to something like CMAKe which is only a tool and doesn't link to the package).

> if the 'nompi' version is the default anyway

There is no "default" There is only what the environment solver finds to be the first package which meets the requirements.

> if removing the condition just make things more reliable

By removing `{% if mpi != 'nompi' %}` you are saying that libraries compiled against libvdwxc-nompi must be run with libvdwxc-nompi. As opposed to *any* libvdwxc. Is that what you want?

My suggested change only adds `- {{ pin_subpackage('libvdwxc', max_pin='x.x') }}` because I think that compiling against libvdwc of any *variant* requires libvdwc of the same version *number* (not variant). 

tl;dr If it's OK to use libvdwc-mpich when you compiled against libvdwc-nompi, then put the `{% if mpi != 'nompi' %}` back.

---

All this makes sense, thanks for the explanation. 

Version number is important, therefore pinning was a good idea.

We should be in the case you described in the last paragraph (for compilation only). At runtime, if the calling program has no MPI awareness or vice-versa, things will definitely go wrong so I'm keen on leaving the condition out.

---

Thanks for your patience.